tests/json.ml

```diff
@@ -5,13 +5,122 @@ let good = "
  { \"a\":\"b\" }
 "
 
-let _ =
-  Printf.printf "Parsing good JSON ... %!";
-  let _ = Jsonrpc.of_string good in
+let v1 = "{
+  \"method\": \"session.login_with_password\",
+	\"params\": [\"user\", \"password\"],
+	\"id\": 0
+}"
+
+let v1_null_id = "{
+  \"method\": \"session.login_with_password\",
+	\"params\": [\"user\", \"password\"],
+	\"id\": null
+}"
+
+let v1_bad_id = "{
+	\"method\": \"session.login_with_password\",
+	\"params\": [\"user\", \"password\"],
+	\"id\": \"0\"
+}"
+
+let v1_no_params ="{
+	\"method\": \"session.login_with_password\",
+	\"id\": \"0\"
+}"
+
+let v1_no_id = "{
+	\"method\": \"session.login_with_password\",
+	\"params\": [\"user\", \"password\"]
+}"
+
+let v2 ="{
+	\"method\": \"session.login_with_password\",
+	\"params\": [\"user\", \"password\"],
+	\"id\": 0,
+	\"jsonrpc\": \"2.0\"
+}"
+
+let v2_no_params = "{
+	\"method\": \"session.login_with_password\",
+	\"id\": 0,
+	\"jsonrpc\": \"2.0\"
+}"
+
+let v2_null_id = "{
+	\"method\": \"session.login_with_password\",
+  \"params\": [\"user\", \"password\"],
+  \"id\": null,
+	\"jsonrpc\": \"2.0\"
+}"
+
+let v2_no_method = "{
+	\"params\": [\"user\", \"password\"],
+	\"id\": 0,
+	\"jsonrpc\": \"2.0\"
+}"
+
+let v2_bad_id = "{
+	\"method\": \"session.login_with_password\",
+	\"params\": [\"user\", \"password\"],
+	\"id\": \"0\",
+	\"jsonrpc\": \"2.0\"
+}"
 
+let v2_bad_jsonrpc = "{
+	\"method\": \"session.login_with_password\",
+	\"params\": [\"user\", \"password\"],
+	\"id\": 0,
+	\"jsonrpc\": \"1.0\"
+}"
+
+let tests1 = [
+  "bad", bad, false;
+  "good", good, true;
+]
+
+let tests2 = [
+  "v1", v1, true;
+  "v1_null_id", v1_null_id, false; (*stricter than the specs*)
```

Comment:
I considered changing them in this PR (the sooner the better), but @robhoes recommended to leave it as is. To my understanding we don't have a use case for them.